x-parser.js

```diff
@@ -1062,13 +1061,18 @@ export class XParser {
             }
           }
           stringIndex = nextStringIndex; // Update out pointer from our pattern match.
+          nextStringIndex = null;
```

Comment:
Null-out the `nextStringIndex` (i.e., `end`) for the next loop. Now that we report it, we want it to be roughly correct.